### PR TITLE
Only return matched results for App filters

### PR DESCRIPTION
**Pull request by benjaminpkane**
Changes:
  * Rename `Show no value/confidence` checkboxes to `Filter no value/confidence`
  * The Fields Sidebar label filters now exclude samples that do not have at least one matching label in their result, including frame labels 

Screenshot from 2021-01-19 14-39-22
Screenshot from 2021-01-19 14-37-55
Screenshot from 2021-01-19 14-37-38

### CODE REVIEW

**Comment by brimoor** [comment]
Okay so here's the code snippet that shows an empty App when the view does contain stuff:

```py
import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F

dataset2 = foz.load_zoo_dataset(
    "coco-2017",
    split="validation",
    max_samples=100,
    shuffle=True,
)

view2 = dataset2.filter_labels("ground_truth", F("label") == "person", only_matches=True)

session = fo.launch_app(view=view2)  # shows empty view but `view2` has stuff in it
```

**Comment by benjaminpkane** [comment]
> ```python
> import fiftyone as fo
> import fiftyone.zoo as foz
> from fiftyone import ViewField as F
> 
> dataset2 = foz.load_zoo_dataset(
>     "coco-2017",
>     split="validation",
>     max_samples=100,
>     shuffle=True,
> )
> 
> view2 = dataset2.filter_labels("ground_truth", F("label") == "person", only_matches=True)
> 
> session = fo.launch_app(view=view2)
> ```

I've tested single labels and list labels for sample and frame labels. Results are correct for me now.

File: fiftyone/core/stages.py
Comment (brimoor) [inline fiftyone/core/stages.py:1104]: I must admit that I'm a bit intimidated by the `hide_frames`, `hide_results`, `squash_frames`, `include_frames`, `_needs_frames` business in the codebase at the moment. I don't quite understand it all.

So for now I'll just verify that the API works :)

**Review by brimoor — COMMENTED** [review]
This is so cool to use! 💯 💯 💯 

Works like a charm on image datasets. Just need to do a bit more testing on video datasets before I can approve the scope here.

**Comment by brimoor** [comment]
I used this feature a bit more and left my enhancement suggestions in this issue https://github.com/voxel51/fiftyone/issues/803.

Feel free to discuss there, and of course triage them in separate PRs if you prefer. But I thought I'd mention them here since you're in this code right now :)

I'll approve this PR once I test video a bit more.

**Review by brimoor — APPROVED** [review]
LGTM. Awesome stuff :)

**Pull request merged**
